spigot/components/src/main/kotlin/com/mineinabyss/geary/minecraft/systems/BossBarDisplaySystem.kt

```diff
@@ -0,0 +1,39 @@
+package com.mineinabyss.geary.minecraft.systems
+
+import com.mineinabyss.geary.ecs.api.systems.TickingSystem
+import com.mineinabyss.geary.ecs.engine.iteration.QueryResult
+import com.mineinabyss.geary.ecs.query.Query
+import com.mineinabyss.geary.minecraft.components.DisplayBossBar
+import org.bukkit.Bukkit
+import org.bukkit.Location
+import org.bukkit.entity.Player
+
+/**
+ * Handles displaying of boss bars.
+ */
+public object BossBarDisplaySystem : TickingSystem(interval = 10) {
```

Comment:
The system itself is a query, you can just add
```kotlin
private val QueryResult.bossbar by get<DisplayBossBar>()
```
as a property inside of it and access it within `QueryResult.tick()` (p.s. that function runs for each entity so right now you are essentially matching against literally all entities and then every time running a foreach on all the things in `DisplayBossBarQuery`)